magenta-lib/src/test/scala/magenta/input/DeploymentResolverTest.scala

```diff
@@ -0,0 +1,302 @@
+        "testTemplate" -> deploymentTemplate("nestedTemplate").withDependencies("template-dep")
+      )),
+      Map(
+        "nested-dep" -> deploymentType("autoscaling"),
+        "template-dep" -> deploymentType("autoscaling"),
+        "deployment-dep" -> deploymentType("autoscaling"),
+        "test" -> deploymentTemplate("testTemplate").withDependencies("deployment-dep")
+      )
+    )
+    val deployments = assertDeployments(DeploymentResolver.resolve(yaml))
+    deployments.size should be (4)
+    val deployment = deployments.find(_.name == "test").get
+    deployment.dependencies should be(List("deployment-dep"))
+  }
+
+  it should "correctly prioritise dependencies from template when not specified in deployment" in {
+    val yaml = RiffRaffYaml(
+      Some(List("global-stack")), None,
+      Some(Map(
+        "nestedTemplate" -> deploymentType("testType").withDependencies("nested-dep"),
+        "testTemplate" -> deploymentTemplate("nestedTemplate").withDependencies("template-dep")
+      )),
+      Map(
+        "nested-dep" -> deploymentType("autoscaling"),
+        "template-dep" -> deploymentType("autoscaling"),
+        "test" -> deploymentTemplate("testTemplate")
+      )
+    )
+    val deployments = assertDeployments(DeploymentResolver.resolve(yaml))
+    deployments.size should be (3)
+    val deployment = deployments.find(_.name == "test").get
+    deployment.dependencies should be(List("template-dep"))
+  }
+
+  it should "correctly prioritise dependencies from nested template when not specified in deployment or template" in {
+    val yaml = RiffRaffYaml(
+      Some(List("global-stack")), None,
+      Some(Map(
+        "nestedTemplate" -> deploymentType("testType").withDependencies("nested-dep"),
+        "testTemplate" -> deploymentTemplate("nestedTemplate")
+      )),
+      Map(
+        "nested-dep" -> deploymentType("autoscaling"),
+        "test" -> deploymentTemplate("testTemplate")
+      )
+    )
+    val deployments = assertDeployments(DeploymentResolver.resolve(yaml))
+    deployments.size should be (2)
+    val deployment = deployments.find(_.name == "test").get
+    deployment.dependencies should be(List("nested-dep"))
+  }
+
+  it should "report an error if a named template doesn't exist" in {
+    val yaml = RiffRaffYaml(
+      Some(List("global-stack")), None,
+      Some(Map(
+        "nestedTemplate" -> deploymentType("testType").withDependencies("nested-dep"),
+        "testTemplate" -> deploymentTemplate("nestedTemplate")
+      )),
+      Map("test" -> deploymentTemplate("nonExistentTemplate"))
+    )
+    val deployments = DeploymentResolver.resolve(yaml)
+    deployments.size should be (1)
+    deployments.head.isLeft should be(true)
+    val errors = deployments.head.left.get
+    errors should be(List("Template with name nonExistentTemplate (specified in test) does not exist"))
+  }
+
+  it should "report missing dependencies" in {
```

Comment:
As with the logic that produces the errors, this seems to be testing the same thing as the above test?